Sources/GRPCHTTP2Core/Server/HTTP2ServerTransport.swift

```diff
@@ -0,0 +1,174 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import GRPCCore
+import NIOHTTP2
+
+/// A namespace for the HTTP/2 server transport.
+public enum HTTP2ServerTransport {}
+
+extension HTTP2ServerTransport {
+  /// A namespace for HTTP/2 server transport configuration.
+  public enum Config {}
+}
+
+extension HTTP2ServerTransport.Config {
+  public struct Compression: Sendable {
+    /// Compression algorithms enabled for inbound messages.
+    ///
+    /// - Note: ``CompressionAlgorithm/none`` is always supported, even if it isn't set here.
+    public var enabledAlgorithms: CompressionAlgorithmSet
+
+    /// Creates a new compression configuration.
+    ///
+    /// - SeeAlso: ``defaults``.
+    public init(enabledAlgorithms: CompressionAlgorithmSet) {
+      self.enabledAlgorithms = enabledAlgorithms
+    }
+
+    /// Default values, compression is disabled.
+    public static var defaults: Self {
+      Self(enabledAlgorithms: .none)
+    }
+  }
+
+  @available(macOS 13.0, iOS 16.0, watchOS 9.0, tvOS 16.0, *)
+  public struct Keepalive: Sendable {
+    /// The amount of time to wait after reading data before sending a keepalive ping.
+    public var time: Duration
+
+    /// The amount of time the server has to respond to a keepalive ping before the connection is closed.
+    public var timeout: Duration
+
+    /// Whether the server allows the client to send keepalive pings when there are no calls in progress.
+    public var permitWithoutCalls: Bool
+
+    /// The minimum allowed interval the client is allowed to send keep-alive pings.
+    /// Pings more frequent than this interval count as 'strikes' and the connection is closed if there are
+    /// too many strikes.
+    public var minPingIntervalWithoutCalls: Duration
+
+    /// Creates a new keepalive configuration.
+    public init(
+      time: Duration,
+      timeout: Duration,
+      permitWithoutCalls: Bool,
+      minPingIntervalWithoutCalls: Duration
+    ) {
+      self.time = time
+      self.timeout = timeout
+      self.permitWithoutCalls = permitWithoutCalls
+      self.minPingIntervalWithoutCalls = minPingIntervalWithoutCalls
+    }
+
+    /// Default values. The time after reading data a ping should be sent defaults to 2 hours, the timeout for
+    /// keepalive pings defaults to 20 seconds, pings are not permitted when no calls are in progress, and
+    /// the minimum allowed interval for clients to send pings defaults to 5 minutes.
+    public static var defaults: Self {
+      Self(
+        time: .seconds(2 * 60 * 60),  // 2 hours
+        timeout: .seconds(20),
+        permitWithoutCalls: false,
+        minPingIntervalWithoutCalls: .seconds(5 * 60)  // 5 minutes
+      )
+    }
+  }
+
+  @available(macOS 13.0, iOS 16.0, watchOS 9.0, tvOS 16.0, *)
+  public struct Idle: Sendable {
+    /// The maximum amount of time a connection may be idle before it's closed.
+    public var maxTime: Duration?
```

Comment:
Should we roll this up into the `Connection` config? `connection.maxIdleTime = <whatever>` seems quite natural, `idle.maxTime = <whatever>` feels a little less so. WDYT?